chef/cookbooks/provisioner/templates/default/autoyast-upgrade.xml.erb

```diff
@@ -0,0 +1,171 @@
+      </listentry>
+    <% end %>
+    </add_on_products>
+  </add-on>
+  <timezone>
+    <hwclock>UTC</hwclock>
+    <timezone><%=@timezone%></timezone>
+  </timezone>
+  <services-manager>
+    <services>
+      <enable config:type="list">
+        <service>sshd</service>
+      </enable>
+    </services>
+  </services-manager>
+  <software>
+    <packages config:type="list">
+      <package>autoyast2-installation</package>
+<% if @architecture == "x86_64" -%>
+      <package>biosdevname</package>
+<% end -%>
+      <package>netcat</package>
+      <package>ruby2.1-rubygem-chef</package>
+<% if @is_ses -%>
+      <package>ses-release</package>
+<% else -%>
+      <package>suse-openstack-cloud-release</package>
+      <package>supportutils-plugin-suse-openstack-cloud</package>
+<% end -%>
+<% @packages.each do |package| -%>
+      <package><%= package %></package>
+<% end -%>
+    </packages>
+    <scripts>
+      <chroot-scripts config:type="list">
+        <script>
+          <chrooted config:type="boolean">false</chrooted>
+          <debug config:type="boolean">true</debug>
+          <filename>crowbar_post</filename>
+          <source>
+            <![CDATA[
+            export PS4='${BASH_SOURCE}@${LINENO}(${FUNCNAME[0]}): '
+            HTTP_SERVER="<%= @admin_node_ip %>:<%= @web_port %>"
+            IP="<%= @admin_node_ip %>"
+            HOSTNAME="<%= @node_fqdn %>"
+
+            key_re='crowbar\.install\.key=([^ ]+)'
+            if [[ $(cat /proc/cmdline) =~ $key_re ]]; then
+                export CROWBAR_KEY="${BASH_REMATCH[1]}"
+                echo "$CROWBAR_KEY" >/mnt/etc/crowbar.install.key
+            elif [[ -f /mnt/etc/crowbar.install.key ]]; then
+                export CROWBAR_KEY="$(cat /mnt/etc/crowbar.install.key)"
+            fi
+            post_state() {
+                mkdir -p /var/log/crowbar
+                local curlargs=(-o "/mnt/var/log/crowbar/$1-$2.json" --connect-timeout 60 -s \
+                  -L -X POST --data-binary "{ \"name\": \"$1\", \"state\": \"$2\" }" \
+                  -H "Accept: application/json" -H "Content-Type: application/json" \
+                  --max-time 240)
+                [[ $CROWBAR_KEY ]] && curlargs+=(-u "$CROWBAR_KEY" --digest --anyauth)
+                curl "${curlargs[@]}" "http://$IP/crowbar/crowbar/1.0/transition/default"
+            }
+
+            mkdir -p /mnt/root/.ssh
+            chmod 700 /mnt/root/.ssh
+
+            if ! curl -s -o /mnt/root/.ssh/authorized_keys.wget \
+                http://$HTTP_SERVER/authorized_keys ||\
+                grep -q "Error 404" /mnt/root/.ssh/authorized_keys.wget; then
+                rm -f /mnt/root/.ssh/authorized_keys.wget
+            else
+                chmod 644 /mnt/root/.ssh/authorized_keys
+                cat /mnt/root/.ssh/authorized_keys.wget >> /mnt/root/.ssh/authorized_keys
+                rm -f /mnt/root/.ssh/authorized_keys.wget
+            fi
```

Comment:
Doesn't really matter, but more out of curiosity: this bit (for authorized keys) should not be needed, since the old file is still there, right?